doc/plugin_server_nodeattestor_gcp_iit.md

```diff
@@ -42,4 +44,5 @@ If `use_instance_metadata` is true, then the Google Compute Engine API is querie
 
 ## Authenticating with the Google Compute Engine API
 
-The plugin uses the Application Default Credentials to authenticate with the Google Compute Engine API, as documented by [Setting Up Authentication For Server to Server](https://cloud.google.com/docs/authentication/production).
+The plugin uses the Application Default Credentials to authenticate with the Google Compute Engine API, as documented by [Setting Up Authentication For Server to Server](https://cloud.google.com/docs/authentication/production). It is recommended that a non default Service Account with `compute.instances.get
+` permission and scope allowing access to Google Compute Engine API be associated with the GCP instance when running the SPIRE server on GCP. When SPIRE Server is not running on a GCP instance, a Service Account key file may be specified either by setting the `GOOGLE_APPLICATION_CREDENTIALS` environment variable or by setting the `service_account_file` configuration.
```

Comment:
Hmm, this wording is pretty specific with the IAM role but not the scope. Maybe it shouldn't spell out exact permissions and scopes and instead link to the google documentation? Something like:

The plugin uses the Application Default Credentials to authenticate with the Google Compute Engine API, as documented by [Setting Up Authentication For Server to Server](https://cloud.google.com/docs/authentication/production). When SPIRE Server is running inside GCP, it will use the default service account credentials available to the container it is running under. When running outside GCP, or if non-default credentials are needed, the path to the service account file containing the credentials may be specified using the `GOOGLE_APPLICATION_CREDENTIALS` environment variable or the `service_account_file` configurable (see [Configuration](#configuration)).

The service account must have IAM permissions and Authorization Scopes granting access to the following APIs:
* [instances.get](https://cloud.google.com/compute/docs/reference/rest/v1/instances/get)